### PR TITLE
remove dependency on DockerNAT for Linux containers on Windows

### PR DESCRIPTION
**Pull request by veyalla**
Docker Desktop has removed 'DockerNAT' interface in its newer versions. 

This change removes the Windows install script dependency on the interface allowing use of Linux containers on Windows with newer Docker Desktop versions. Though not tested, this change should work with older versions of Docker Desktop as well.

### CODE REVIEW

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 1]
```diff
@@ -1949,15 +1949,17 @@ function Set-ListenConnectUri([string] $ManagementUri, [string] $WorkloadUri) {
     $env:IOTEDGE_HOST = $ManagementUri
 }
 
-function Set-GatewayAddress {
-    $gatewayAddress = (Get-NetIpAddress |
-            Where-Object {$_.InterfaceAlias -like '*vEthernet (DockerNAT)*' -and $_.AddressFamily -eq 'IPv4'}).IPAddress
+function Set-ListenConnectUriForLinuxContainers {
+  # "host.docker.internal" is a well-known address that maps to the Host from inside a container on Docker Desktop.
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: This should be four-space indentation like the rest of the file.

**Comment by sandervandevelde** [comment]
Hello, I tried to install IoTEdge using this updated script but I got a not-specific exception after filling in the connectionstring. (I fixed the powershell commands using this commited version)  

I checked the config.yaml file but is was not updated regarding the connection string. 

So this works for me:

```
execute:

. {Invoke-WebRequest -useb https://raw.githubusercontent.com/Azure/iotedge/master/scripts/windows/setup/IotEdgeSecurityDaemon.ps1} | Invoke-Expression; `
Deploy-IoTEdge -ContainerOs Linux


execute:

. {Invoke-WebRequest -useb https://raw.githubusercontent.com/Azure/iotedge/master/scripts/windows/setup/IotEdgeSecurityDaemon.ps1} | Invoke-Expression; `
Initialize-IoTEdge -ContainerOs Linux


execute:

code C:\ProgramData\iotedge\config.yaml


fill in the connection string of your device AGAIN in this section:

provisioning:
  source: "manual"
  device_connection_string: "HostName=edgedemo-ih.azure-devices.net;DeviceId=testps;SharedAccessKey=yadda="


execute:
restart-Service iotedge


execute:

iotedge list


See:

eventually the edgeAgent is made available. You can proceed now
```

Conclusion: the connection string is not inserted in the config.yaml during the Initialize phase.

(I run Docker Desktop 2.2.0.3 (42716) with Linux containers on Windows 10)

**Comment by arsing** [comment]
You're trying to use a script from master with a CAB which is 1.0.8. The CAB and the script from different branches are not expected to be compatible.

**Comment by veyalla** [comment]
@sandervandevelde  I think this is expected. 

The version of the script needs the 1.0.9 version of the iotedged packages installed. Try by providing the [1.0.9-rc5](https://github.com/Azure/azure-iotedge/releases/tag/1.0.9-rc5) cab of iotedged using `-OfflineInstallationPath`  switch (rename the downloaded cab to `Microsoft-Azure-IoTEdge.cab`)

Obviously all this should work out of box when 1.0.9 goes GA.

**Comment by arsing** [comment]
@veyalla You're suggesting to use a script from master with packages from 1.0.9. That is also not expected to work.

**Comment by veyalla** [comment]
Ah, got it, thanks for pointing that out @arsing 

@sandervandevelde I am going to cherry pick this change into 1.0.9 branch. You can then test out when it merges.

**Pull request merged**